clients/ops/admin-ui/src/features/privacy-requests/SortRequestButton.tsx

```diff
@@ -0,0 +1,279 @@
+import { Flex, Icon, IconButton } from "@fidesui/react";
+import React, { useCallback, useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import {
+  clearSortFields,
+  selectPrivacyRequestFilters,
+  setSortDirection,
+  setSortField,
+} from "./privacy-requests.slice";
+
+type UseSortRequestButtonParams = {
+  sortField: string;
+  isLoading: boolean;
+};
+
+export enum ButtonState {
+  ASC = "asc",
+  DESC = "desc",
+  UNSELECTED = "unselected",
+}
+
+const useSortRequestButton = ({
+  sortField,
+  isLoading,
+}: UseSortRequestButtonParams) => {
+  const filters = useSelector(selectPrivacyRequestFilters);
+  const dispatch = useDispatch();
+  const [buttonState, setButtonState] = useState<ButtonState>(
+    ButtonState.UNSELECTED
+  );
+  const [wasButtonJustClicked, setWasButtonJustClicked] = useState(false);
+  useEffect(() => {
```

Comment:
Consolidate the useEffect hooks to one usage.

Reply:
I tested this and it broke the loading spinner. I think that is due to the filters variable changing before `isLoading` was set to true. That causes `setWasButtonJustClicked` to get called before the request actually started. Making so `wasButtonJustClicked` is false and preventing the spinner from showing up.

They could probably still be combined but it makes a little simpler to separate them. What do you think?

Reply:
When you combined them, what were your dependencies listed in the useEffect array?

Reply:
yes, it was `[isLoading, filters]`

Reply:
Perfectly fine to use multiple useEffect hooks. I guess at first glance I was basing this on the component lifecycle. However, managing the component behavior is another approach.